ludwig/models/modules/convolutional_modules.py

```diff
@@ -722,6 +723,152 @@ def call(self, inputs, training=None, mask=None):
         return hidden + shortcut
 
 
+class ResNetBlockLayer(Layer):
+    def __init__(
+        self,
+        num_filters,
+        is_bottleneck,
+        block_fn,
+        num_blocks,
+        strides,
+        weights_regularizer=None,
+        batch_norm_momentum=0.9,
+        batch_norm_epsilon=0.001
+    ):
+        super(ResNetBlockLayer, self).__init__()
+        # Bottleneck blocks end with 4x the number of filters as they start with
+        num_filters_out = num_filters * 4 if is_bottleneck else num_filters
+
+        projection_shortcut = Conv2DLayerFixedPadding(
+            num_filters=num_filters_out,
+            filter_size=1,
+            strides=strides,
+            weights_regularizer=weights_regularizer
+        )
+
+        self.layers = [
+            block_fn(
+                num_filters,
+                strides,
+                weights_regularizer=weights_regularizer,
+                batch_norm_momentum=batch_norm_momentum,
+                batch_norm_epsilon=batch_norm_epsilon,
+                projection_shortcut=projection_shortcut
+            )
+        ]
+
+        for _ in range(1, num_blocks):
+            self.layers.append(
+                block_fn(
+                    num_filters,
+                    1,
+                    weights_regularizer=weights_regularizer,
+                    batch_norm_momentum=batch_norm_momentum,
+                    batch_norm_epsilon=batch_norm_epsilon
+                )
+            )
+
+    def call(self, inputs, training=None, mask=None):
+        hidden = inputs
+
+        for layer in self.layers:
+            hidden = layer(hidden, training=training)
+
+        return hidden
+
+
+class ResNet2(Layer):
+    def __init__(
+        self,
+        resnet_size,
+        is_bottleneck,
+        num_filters,
+        filter_size,
+        conv_stride,
+        first_pool_size,
+        first_pool_stride,
+        block_sizes,
+        block_strides,
+        weights_regularizer=None,
+        batch_norm_momentum=0.9,
+        batch_norm_epsilon=0.001
+    ):
+        super(ResNet2, self).__init__()
+        self.resnet_size = resnet_size
+
+        self.is_bottleneck = is_bottleneck
+        if is_bottleneck:
+            self.block_fn = ResNetBottleneckBlock
```

Comment:
it seems that \block_fn\ is used only in this function so maybe we don't neeed to assign it to self. Also, as it's not a function anymore but a class, maybe let's cal it \block_class\?